### PR TITLE
mds: Reorganize struct members in Mutation header

### PR DESCRIPTION
**Pull request by varshar16**
Fixes: https://tracker.ceph.com/issues/42690
Signed-off-by: Varsha Rao <varao@redhat.com>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by varshar16** [comment]
Rebased.

**Comment by varshar16** [comment]
jenkins test make check arm64

File: src/mds/Mutation.h [hunk 1]
```diff
@@ -436,8 +381,52 @@ struct MDRequestImpl : public MutationImpl {
...
+  Session *session = nullptr;
+  elist<MDRequestImpl*>::item item_session_request;  // if not on list, op is aborted.
+
+  // -- i am a client (master) request
+  cref_t<MClientRequest> client_request; // client request (if any)
+
+  // store up to two sets of dn vectors, inode pointers, for request path1 and path2.
+  vector<CDentry*> dn[2];
+  CDentry *straydn = nullptr;
+  CInode *in[2];
```
Comment (batrick) [inline src/mds/Mutation.h]: `= {}` to 0 initialize.

File: src/mds/Mutation.h [hunk 2]
```diff
@@ -395,17 +337,20 @@ struct MDRequestImpl : public MutationImpl {
...
-    session(NULL), item_session_request(this),
-    client_request(params->client_req), straydn(NULL), snapid(CEPH_NOSNAP),
-    tracei(NULL), tracedn(NULL), alloc_ino(0), used_prealloc_ino(0),
-    internal_op(params->internal_op), internal_op_finish(NULL),
-    internal_op_private(NULL),
-    retry(0),
-    waited_for_osdmap(false), _more(NULL) {
+    item_session_request(this), client_request(params->client_req),
+    internal_op(params->internal_op) {
     in[0] = in[1] = NULL;
```
Comment (batrick) [inline src/mds/Mutation.h]: and then drop this.

File: src/mds/Mutation.h (hunk 1 above)
Comment (varshar16) [reply src/mds/Mutation.h]: Done.

File: src/mds/Mutation.h (hunk 2 above)
Comment (varshar16) [reply src/mds/Mutation.h]: Done.

**Comment by batrick** [comment]
needs rebased again

**Comment by varshar16** [comment]
@batrick rebased.

**Pull request merged**